selfservice/strategy/oidc/provider_gitlab.go

```diff
@@ -0,0 +1,78 @@
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	glapi "github.com/xanzy/go-gitlab"
+	"golang.org/x/oauth2"
+)
+
+var _ = glapi.ProjectClustersService{}
+
+var endpoint = oauth2.Endpoint{
+	TokenURL: "https://gitlab.com/oauth/token",
+	AuthURL:  "https://gitlab.com/oauth/authorize",
+}
+
+type ProviderGitLab struct {
```

Comment:
Wouldn't it make sense to extend the generic OIDC provider we have, like we do for Google?

https://github.com/ory/kratos/blob/master/selfservice/strategy/oidc/provider_google.go

Reply:
@aeneasr I made that change and seems to work great. Thanks!